android/src/main/java/it/innove/Peripheral.java

```diff
@@ -778,38 +756,62 @@ private void processCommands() {
-                Log.d(LOG_TAG, "Remove Notify " + command.getCharacteristicUUID());
-                bleProcessing = true;
-                removeNotify(command.getServiceUUID(), command.getCharacteristicUUID(), command.getCallback());
-            } else if (command.getType() == BLECommand.READ_RSSI) {
-                Log.d(LOG_TAG, "Read RSSI");
-                bleProcessing = true;
-                readRSSI(command.getCallback());
-            } else {
-                // this shouldn't happen
-                throw new RuntimeException("Unexpected BLE Command type " + command.getType());
-            }
-        } else {
-            Log.d(LOG_TAG, "Command Queue is empty.");
+
+        if (command == null) {
+          Log.d(LOG_TAG, "Command Queue is empty.");
+          return;
         }
+
+        try {
+          bleProcessing = true;
+
+          int commandType = command.getType();
+
+          if(commandType == BLECommand.READ){
+            Log.d(LOG_TAG, "$Read " + command.getCharacteristicUUID());
+            lastCommand = "Read";
+            read(command.getServiceUUID(), command.getCharacteristicUUID(), command.getCallback());
+          }
+          else if(commandType == BluetoothGattCharacteristic.WRITE_TYPE_DEFAULT){
+            Log.d(LOG_TAG, "$Write " + command.getCharacteristicUUID());
+            lastCommand = "Write";
+            write(command.getServiceUUID(), command.getCharacteristicUUID(), command.getData(), command.getMaxByteSize(), command.getQueueSleepTime(), command.getCallback(), command.getType());
+          }
+          else if(commandType == BluetoothGattCharacteristic.WRITE_TYPE_NO_RESPONSE){
+            Log.d(LOG_TAG, "$Write No Response " + command.getCharacteristicUUID());
+            lastCommand = "Write No Response";
+            write(command.getServiceUUID(), command.getCharacteristicUUID(), command.getData(), command.getMaxByteSize(), command.getQueueSleepTime(), command.getCallback(), command.getType());
+          }
+          else if(commandType == BLECommand.REGISTER_NOTIFY){
+            Log.d(LOG_TAG, "$Register Notify " + command.getCharacteristicUUID());
+            lastCommand = "Register Notify";
+            registerNotify(command.getServiceUUID(), command.getCharacteristicUUID(), command.getCallback());
+          }
+          else if(commandType == BLECommand.REMOVE_NOTIFY){
+            Log.d(LOG_TAG, "$Remove Notify " + command.getCharacteristicUUID());
+            lastCommand = "Remove Notify";
+            removeNotify(command.getServiceUUID(), command.getCharacteristicUUID(), command.getCallback());
+          }
+          else if(commandType == BLECommand.READ_RSSI){
+            Log.d(LOG_TAG, "$Read RSSI");
+            lastCommand = "Read RSSI";
+            readRSSI(command.getCallback());
+          }
+          else {
+            throw new RuntimeException("Unexpected BLE Command type " + command.getType());
+          }
+
+
+        } catch (Exception e) {
```

Comment:
The main change is here, rather than have the command be responsible for the error callback being invoked, it throws an error which is caught here. This ensures that any errors are handled, and that the native command in the queue is properly completed, rather than prematurely exiting and blocking the queue indefinitely.